kernel: bump 6.6 to 6.6.27

graysky2 opened the pull request:
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.27

All patches automatically rebased.

Build system: x86/64
Build-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, flogic/glinet_gl-mt6000, x86/64/AMD Cezanne
Run-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, flogic/glinet_gl-mt6000, x86/64/AMD Cezanne

graysky2 committed:
kernel: bump 6.6 to 6.6.26

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.26

Manually rebased:
	generic/hack-6.6/210-darwin_scripts_include.patch

Added pending:
	generic/pending-6.6/981-gcc-plugins-stackleak-Avoid-.head.txt.section.patch

All other patches automatically rebased.

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, flogic/glinet_gl-mt6000
Run-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, flogic/glinet_gl-mt6000

Signed-off-by: John Audia <therealgraysky@proton.me>

graysky2 committed:
x86: generic: config-6.6: refresh

Ran `make kernel_menuconfig`

Reference to CONFIG_SPECTRE_BHI_ON=y[1]

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/diff/arch/x86/Kconfig?id=v6.6.26&id2=v6.6.25

Signed-off-by: John Audia <therealgraysky@proton.me>

hardfalcon commented:
Please consider adding [this patch](https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/?id=e7d24c0aa8e678f41457d1304e2091cac6fd1a2e), otherwise bzImage may be unbootable when built with `CONFIG_GCC_PLUGIN_STACKLEAK=y`. See [this thread](https://lore.kernel.org/stable/CAMj1kXFMwxFJJvYaTfUYb88a3iL4VFLWoBVdKj1eUH7q5_fOVw@mail.gmail.com/) for further details.

robimarko commented:
> Please consider adding [this patch](https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/?id=e7d24c0aa8e678f41457d1304e2091cac6fd1a2e), otherwise the kernel bzImage may be unbootable on x86_64 when built with `CONFIG_GCC_PLUGIN_STACKLEAK=y`. See [this thread](https://lore.kernel.org/stable/CAMj1kXFMwxFJJvYaTfUYb88a3iL4VFLWoBVdKj1eUH7q5_fOVw@mail.gmail.com/) for further details.

Is this something new in this point release or it was like this before?

hardfalcon commented:
> Is this something new in this point release or it was like this before?

It's an issue that was introduced in kernel 6.8.5 and most likely also in 6.6.26. I haven't tested 6.6.26, but I ran into the issue with 6.8.5-rc*, but my first report wasn't reproducible by upstream, so they decided to not throw out the commit in question. The fix for the commit is already in the stable queue [for 6.6.27](https://git.kernel.org/pub/scm/linux/kernel/git/stable/stable-queue.git/commit/queue-6.6/gcc-plugins-stackleak-avoid-.head.text-section.patch?id=8718a8b81dc1fa3a672d4ee13018d6bd9e9a0c1b) and [for 6.8.6](https://git.kernel.org/pub/scm/linux/kernel/git/stable/stable-queue.git/commit/queue-6.8/gcc-plugins-stackleak-avoid-.head.text-section.patch?id=29b87e2d7def517709c626cb83a3bd497ab6d52e).

robimarko commented:
> > Is this something new in this point release or it was like this before?
> 
> It's an issue that was introduced in kernel 6.8.5 and most likely also in 6.6.26 (at least [the same commit that caused the issue with 6.8.5](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.8.5&id=6cd1639353e63cc2525bd7b43055a6a61d4d2fdb) is also [in 6.6.26](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.26&id=af90ced75242b5a1ca4fc861f4bd55890a4ed873)). I haven't tested 6.6.26, but I ran into the issue with 6.8.5-rc*, and my first report wasn't reproducible by upstream, so they decided to not throw out the commit in question when releasing 6.6.26 and 6.8.5. The fix for the commit is already in the stable queue [for 6.6.27](https://git.kernel.org/pub/scm/linux/kernel/git/stable/stable-queue.git/commit/queue-6.6/gcc-plugins-stackleak-avoid-.head.text-section.patch?id=8718a8b81dc1fa3a672d4ee13018d6bd9e9a0c1b) and [for 6.8.6](https://git.kernel.org/pub/scm/linux/kernel/git/stable/stable-queue.git/commit/queue-6.8/gcc-plugins-stackleak-avoid-.head.text-section.patch?id=29b87e2d7def517709c626cb83a3bd497ab6d52e).

In that case, please include that commit as well

robimarko commented:
Sorry for the closure, wrong button :)

zxlhhyccc commented:
@graysky2 6.1 kernel no longer updated?

graysky2 commented:
@robimarko - included will push shortly
@zxlhhyccc - I have all my builds using the test kernel now so no need for 6.1 on my boxes

graysky2 committed:
kernel: bump 6.6 to 6.6.27

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.27

Removed upstreamed:
	pending-6.6/981-gcc-plugins-stackleak-Avoid-.head.txt.section.patch[1]

All patches automatically rebased.

1. https://github.com/gregkh/linux/commit/9dff96b8b3a404d3b3581af46450f2efeb7d290a

Build system: x86/64
Build-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64/AMD Cezanne
Run-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64/AMD Cezanne

Signed-off-by: John Audia <therealgraysky@proton.me>

vincejv commented:
close and skip to 6.6.27 released a few hours ago

hauke commented on target/linux/generic/pending-6.6/981-gcc-plugins-stackleak-Avoid-.head.txt.section.patch:
```diff
@@ -20,9 +20,9 @@ Signed-off-by: Kees Cook <keescook@chromium.org>
+@@ -469,6 +469,8 @@ static bool stackleak_gate(void)
  			return false;
- 		if (STRING_EQUAL(section, ".entry.text"))
+ 		if (STRING_EQUAL(section, ".head.text"))
```
Please remove this patch it was added to v6.6.27 in https://github.com/gregkh/linux/commit/9dff96b8b3a404d3b3581af46450f2efeb7d290a

graysky2 replied:
Done!
